assets/js/modules/pagespeed-insights/components/dashboard/DashboardPageSpeed.js

```diff
@@ -229,28 +229,30 @@ export default function DashboardPageSpeed() {
 			const allAudits = select(
 				MODULES_PAGESPEED_INSIGHTS
 			).getAuditsWithStackPack( referenceURL, strategy, 'wordpress' );
+
 			if ( ! allAudits || ! Object.keys( allAudits ).length ) {
 				return [];
 			}
 
-			const audits = [];
-			Object.keys( allAudits ).forEach( ( auditSlug ) => {
-				const audit = allAudits[ auditSlug ];
-				if (
-					( audit.scoreDisplayMode !== 'numeric' &&
-						audit.scoreDisplayMode !== 'binary' ) ||
-					audit.score >= 0.9
-				) {
-					return;
-				}
-
-				audits.push( {
-					id: audit.id,
-					title: audit.title,
-				} );
-			} );
-
-			return audits;
+			return Object.keys( allAudits )
+				.filter(
+					( auditSlug ) =>
+						allAudits[ auditSlug ].scoreDisplayMode ===
+							'metricSavings' && allAudits[ auditSlug ].score < 1
+				)
+				.sort( ( a, b ) => {
+					// If the scores are the same, sort alphabetically by
+					// audit slug. This is how the API returns audits.
+					if ( allAudits[ a ].score === allAudits[ b ].score ) {
+						return a.localeCompare( b );
+					}
```

Comment:
According to the ACs:

> The secondary order should be the order returned by the API.

The API returns the audits as an object with alphabetically ordered keys. However, when we process (filter) this object inside the `getAuditsWithStackPack` selector with `Object.keys`, this order is lost.

To ensure the AC is met, we order audits here alphabetically.

Reply:
Thanks @nfmohit, good spot!